python/lsst/daf/butler/_limited_butler.py

```diff
@@ -91,6 +132,56 @@ def putDirect(self, obj: Any, ref: DatasetRef) -> DatasetRef:
         """
         raise NotImplementedError()
 
+    def get(
+        self,
+        ref: DatasetRef,
+        /,
+        *,
+        parameters: dict[str, Any] | None = None,
+        storageClass: StorageClass | str | None = None,
+    ) -> Any:
+        """Retrieve a stored dataset.
+
+        Parameters
+        ----------
+        ref: `DatasetRef`
+            A resolved `DatasetRef` directly associated with a dataset.
+        parameters : `dict`
+            Additional StorageClass-defined options to control reading,
+            typically used to efficiently read only a subset of the dataset.
+        storageClass : `StorageClass` or `str`, optional
+            The storage class to be used to override the Python type
+            returned by this method. By default the returned type matches
+            the dataset type definition for this dataset. Specifying a
+            read `StorageClass` can force a different type to be returned.
+            This type must be compatible with the original type.
```

Comment:
I'd rather not have a `storageClass` argument here, since the caller can override the registry storage class via `ref.datasetType.storageClass` already and I think that makes it more confusing than convenient (especially since right now `LimitedButler` mostly exists just for task execution and that doesn't use this parameter at all).  I think that argument is even more applicable to `Datastore.get`, since that really isn't in need of convenience functionality since only butler should call it, but that may be out of scope for this ticket.

To be clear, I _do_ think it's worth having a `storageClass` parameter to full `Butler.get`, since there it's very convenient when not passing a `DatasetRef`, and then since it'll present anyway it might as well further override what's in the ref when a `DatasetRef` is passed.

Reply:
My main argument for keeping it is that it's forwarded directly to datastore and datastore will use it and treat it as the priority. It seems odd to declare that the base class can't support the parameter even when it can.